### ISSUE TITLE
Hot Reload Not Working

### ISSUE THREAD

**Issue by SDVII** [nuxt/typescript#108]
**Describe the bug**
Hot-reload is not consistently working when changes are being made to external `.ts` files.

**To Reproduce**
1. Create a Nuxt project.
2. Add TS support.
3. Create a new folder, let's name it services.
4. Create a new `helloWorld.ts` file in it and write some lines.
5. Run `npm run dev`.
6. Make new changes.
7. Find that changes were not applied till the npm command is restarted.

**Expected behavior**
The new changes on external `.ts` files should trigger a hot-reload.

**Additional context**
The hot-reloads works inconsistently, so multiple changes might be needed to reproduce the issue.

**Comment by kevinmarrec** [comment]
@SDVII Where is your `helloWorld.ts`  used ?

**Comment by SDVII** [comment]
I used as an external script that I call from one of my `.vue` files.
I placed in a folder called `app` in the root directory.
The hot reload is working fine when I update the Vue files but not for the external file in the `app` folder.

**Comment by kevinmarrec** [comment]
@SDVII I think it's normal behavior, I'm not sure Nuxt look for all imports in `.vue` files to make them watched. Doesn't sound related to TypeScript unless you can confirm it works when the file is JavaScript and not TypeScript.

**Comment by SDVII** [comment]
A weird thing just happened.

I created a new folder and called it `src`, and I placed a new  `.ts` file there. When I edited the file, the hot reload was triggered unlike the one in the `app` folder.

After some testing, it seems that any file that is not in the `app` folder is triggering the hot reload and the ones in it don't. I made sure the `tsconfig.json` had no special rules for that folder, but nothing seems to indicate so. Is the `app` folder reserved or something?

This is my copy of the `tsconfig.js`

```JSON
{
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        "lib": [
            "esnext",
            "esnext.asynciterable",
            "dom"
        ],
        "esModuleInterop": true,
        "allowJs": true,
        "sourceMap": true,
        "strict": true,
        "noEmit": true,
        "baseUrl": ".",
        "outDir": "./dist",
        "paths": {
            "~/*": [
                "./*"
            ],
            "@/*": [
                "./*"
            ]
        },
        "types": [
            "@types/node",
            "@nuxt/types"
        ]
    },
    "exclude": [
        "node_modules"
    ]
}
```

`nuxt.config.js`:

```JS
export default {
  mode: 'spa',
  head: {
    title: process.env.npm_package_name || '',
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      {
        hid: 'description',
        name: 'description',
        content: process.env.npm_package_description || '',
      },
    ],
    link: [{ rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' }],
  },
  loading: { color: '#fff' },
  css: [],
  env: {},
  plugins: [],
  buildModules: ['@nuxt/typescript-build'],
  modules: [],
  build: {
    extend(config, ctx) {},
  },
};
```

**Comment by kevinmarrec** [comment]
I know that the `app` is kind of reserved since some feature implemented around Nuxt 2.9 that allows people to override some templates used by Nuxt or external modules. But still I don't think is really related. I would first try https://nuxtjs.org/api/configuration-watch in the first case see how it goes.

**Comment by SDVII** [comment]
Alright, I'll look into it. Thanks for the replies and keep up the great work!
